dogpile/cache/region.py

```diff
@@ -1053,7 +1064,12 @@ def async_creator(key, mutex):
                 # sort the keys, the idea is to prevent deadlocks.
                 # though haven't been able to simulate one anyway.
                 keys_to_get = sorted(mutexes)
+                start_time = time.time()
                 new_values = creator(*keys_to_get)
+                log.debug(
+                    'Generate cache in %.2f s. for key: %r',
```

Comment:
**Hervé Beraud** (4383) wrote:

IMHO, I think it's most useful to print here the cache creation date and time instead of the used time to generate the cache.

This operation will surely only take just few miliseconds to generate this cache... In other words this information isn't reallly useful.

Also, I'm not sure, I do not have related benchmark, but I guess `%.2f` isn't enough to represent a quick treatment like this, possibly you will to only display "0.00 s" the most of the time.

As a user of this module I prefer know when my cache was created rather than the time used to generate it.

Thoughts?

Reply:
this could be a slow query against a database or web service and can take any arbitrary amount of time, it could be minutes or any amount of time, which is useful to see in the logs as it would describe why a request is hanging.  or if the async runner is used it can take minutes or hours while the app continues to run normally.  dogpile.cache's use case is for optimizing data fetches that may be slow, it's the most important number to see in the logs really.

Reply:
Make sense

Reply:
also the log line will have the timestamp, so that's the time in which the finished value was created

Reply:
Good point :)